contextily/tile.py

```diff
@@ -162,10 +171,30 @@ def bounds2img(w, s, e, n, zoom='auto',
     return merged, extent
 
 
-def _fetch_tile(tile_url, wait, max_retries):
+def _fetch_tile(tile_url, wait, max_retries, cachedir=TILE_CACHE_DIR):
+    cached_path = None
+    if cachedir is not None:
+        if not os.path.exists(cachedir):
+            os.makedirs(cachedir)
+        
+        
+        try:
+            # Split of URL header and make sure path is directorified
+            cached_path = os.path.join(cachedir, 
+                os.path.sep.join(tile_url.split("://")[1:]).replace("/", os.path.sep))
+            image = Image.open(cached_path).convert('RGB')
+            image = np.asarray(image)
+            return image
+        except (IOError, OSError, FileNotFoundError) as e:
+            pass
+
     request = _retryer(tile_url, wait, max_retries)
     with io.BytesIO(request.content) as image_stream:
         image = Image.open(image_stream).convert('RGB')
+        if cached_path:
+            if not os.path.exists(os.path.dirname(cached_path)):
```

Comment:
The `os.path.exists` check might not be required in Python 3, but I am used to having to use it from Python 2.